### PR TITLE
Add Discv5.1 bootstrap nodes for Zinken

### PR DESCRIPTION
**Pull request by zah**
Also prepares the Makefiles for supporting Medalla through a separate binary

### CODE REVIEW

**Review by arnetheduck — APPROVED** [review]

**Comment by arnetheduck** [comment]
the difference between 1.0 and 0.12 doesn't look that bloody - @tersec wdyt of simply adding it to the runtime presets so we don't need another binary? what's the impact?

**Review by tersec — APPROVED** [review]

**Comment by tersec** [comment]
> the difference between 1.0 and 0.12 doesn't look that bloody - @tersec wdyt of simply adding it to the runtime presets so we don't need another binary? what's the impact?

That was my first approach. It'd have been preferable, of course.

However, `EPOCHS_PER_ETH1_VOTING_PERIOD` changes from 32 to 64 between v0.12.3 and 1.0.0-rc.0:
https://github.com/ethereum/eth2.0-specs/blob/7748c70c15b030ca42ad9fa94cdb96d3d9e3e6e6/configs/mainnet/phase0.yaml#L98
https://github.com/ethereum/eth2.0-specs/blob/a64ae8137d4b509b0cceff97ca8499d2ce13214d/configs/mainnet/phase0.yaml#L94

It's used as part of the SSZ `List` in https://github.com/ethereum/eth2.0-specs/blob/v1.0.0-rc.0/specs/phase0/beacon-chain.md#beaconstate:
`eth1_data_votes: List[Eth1Data, EPOCHS_PER_ETH1_VOTING_PERIOD * SLOTS_PER_EPOCH]`

Which we render as
https://github.com/status-im/nimbus-eth2/blob/d98be078c317d21eaec4e8bb33f626c6a54287f1/beacon_chain/spec/datatypes.nim#L309-L310

Which currently requires a compile-time constand size:
https://github.com/status-im/nimbus-eth2/blob/6cf7e837ceb6cfb764b3d95a71fc3b31e2c7a2d3/beacon_chain/ssz/types.nim#L79

I proposed was having a modified `HashList` type which might not be as optimal, but would allow runtime (via runtime presets) modification and implement what you're pointing broadly towards, but my initial goal was to get 1.0.0-rc.0 into `devel` so people could start playing withit per EF's recommendation, so I did the most expedient thing.

I'm fine with this PR, for similar reasons. None of them create significant tech debt that's not easy to remove if/when we don't need multiple binaries. Furthermore, in anticipation of hard forks in the future in Eth2, we're likely going to have to become more runtime-defined, and give up some of that compile-time optimality.

**Comment by zah** [comment]
Two CI runs on AMD64 macOS has timed out which seems like a temporary glitch on Travis's side.

File: Makefile [hunk 1]
```diff
@@ -312,17 +313,17 @@ clean-medalla:
 ### zinken
 ###
 zinken: | beacon_node signing_process
-	$(call CONNECT_TO_NETWORK,zinken)
+	$(call CONNECT_TO_NETWORK,zinken,beacon_node)
```
Comment (kdeme) [inline Makefile:316]: Not sure it is fine to make zinken use `UseDiscv51`

File: beacon_chain/beacon_node.nim.cfg [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 # Use only `secp256k1` public key cryptography as an identity in LibP2P.
 -d:"libp2p_pki_schemes=secp256k1"
+-d:"UseDiscv51=true"
```
Comment (kdeme) [inline beacon_chain/beacon_node.nim.cfg:7]: You still actually needed to do the specific imports in eth2discovery. Sorry the flag was not really useful, improved here: https://github.com/status-im/nim-eth/pull/305

**Pull request merged**